### PR TITLE
Use default spacy lemmatizer

### PR DESCRIPTION
**Pull request by ayush-1506**
Fixes: #715

### CODE REVIEW

**Comment by ayush-1506** [comment]
Results:

Before changes:
```
Recall @ 1: 35.43689320388349%
Recall @ 5: 38.83495145631068%
Recall @ 10: 39.07766990291262%
Precision @ 1: 0.546448087431694%
Precision @ 5: 7.431693989071022%
Precision @ 10: 4.398907103825126%
Recall: 39.07766990291262%
Precision: 4.398907103825136%
MAP@k : 15.2855841790268%
```

After changes:
```
Recall @ 1: 31.55339805825243%
Recall @ 5: 34.95145631067961%
Recall @ 10: 35.19417475728155%
Precision @ 1: 0.546448087431694%
Precision @ 5: 6.612021857923482%
Precision @ 10: 3.961748633879772%
Recall: 35.19417475728155%
Precision: 3.9617486338797816%
MAP@k : 13.97627721398213%
```

**Comment by marco-c** [comment]
Since it appears to be worse, let's make it an option.

File: bugbug/similarity.py [hunk 1]
```diff
@@ -78,21 +78,22 @@ def get_text(self, bug):
...
 
         text = re.sub("[^a-zA-Z0-9]", " ", text)
 
-        ps = PorterStemmer()
+        text = [word.lemma_ for word in nlp(text)]
 
-        tokenized_text = (
-            word_tokenize(text.lower()) if self.nltk_tokenizer else text.lower().split()
-        )
+        ps = PorterStemmer()
```
Comment (marco-c) [inline bugbug/similarity.py]: When we lemmatize, we should not stem.

File: bugbug/similarity.py [hunk 2]
```diff
@@ -78,22 +79,29 @@ def __init__(self, cleanup_urls=True, nltk_tokenizer=False):
     def get_text(self, bug):
         return "{} {}".format(bug["summary"], bug["comments"][0]["text"])
 
-    def text_preprocess(self, text, join=False):
+    def text_preprocess(self, text, spacy_lemmatization=False, join=False):
+        nlp = spacy.load("en_core_web_sm")
```
Comment (marco-c) [inline bugbug/similarity.py]: It'd be better to do the loading only once (at the top of the file in a global variable so you can use it everywhere), and not for every call to text_preprocess

File: bugbug/similarity.py [hunk 3]
```diff
@@ -78,22 +79,29 @@ def __init__(self, cleanup_urls=True, nltk_tokenizer=False):
     def get_text(self, bug):
         return "{} {}".format(bug["summary"], bug["comments"][0]["text"])
 
-    def text_preprocess(self, text, join=False):
+    def text_preprocess(self, text, spacy_lemmatization=False, join=False):
```
Comment (marco-c) [inline bugbug/similarity.py]:
```suggestion
    def text_preprocess(self, text, lemmatization=False, join=False):
```

Comment (marco-c) [reply bugbug/similarity.py]: The fact that it is Spacy is just an implementation detail

**Pull request merged**